src/exe-mach-o.c

```diff
@@ -70,6 +77,35 @@ extern void *bxfi_trampoline_end;
     ((uintptr_t) &bxfi_trampoline_end \
     - (uintptr_t) &bxfi_trampoline)
 
+static int mem_protect(void *addr, size_t len, int prot)
+{
+    int result = mprotect(addr, len, prot);
+
+#if defined (HAVE_MACH_VM_PROTECT)
+    if (result == 0)
+        return 0;
+
+    vm_prot_t mach_prot = 0;
+    if (prot & PROT_READ)
+        mach_prot |= VM_PROT_READ;
+    if (prot & PROT_WRITE)
+        mach_prot |= VM_PROT_WRITE;
+    if (prot & PROT_EXEC)
+        mach_prot |= VM_PROT_EXECUTE;
+
+    if ((result = mach_vm_protect(mach_task_self(), (mach_vm_address_t) addr, len, FALSE, mach_prot)) == 0)
```

Comment:
nit: since `result` isn't used, I'd just write this as `if (mach_vm_protect(...) == 0)`

Reply:
Thanks, I'll fix it :)